elrond-wasm-derive/src/generate/abi_gen.rs

```diff
@@ -100,6 +100,52 @@ fn generate_endpoint_snippets(contract: &ContractTrait) -> Vec<proc_macro2::Toke
         .collect()
 }
 
+fn generate_event_snippet(m: &Method, event_name: &str) -> proc_macro2::TokenStream {
+    let event_docs = &m.docs;
+    let input_snippets: Vec<proc_macro2::TokenStream> = m
+        .method_args
+        .iter()
+        .filter_map(|arg| {
+            let mut arg_type = arg.ty.clone();
+            let indexed = arg.metadata.event_topic;
+            clear_all_type_lifetimes(&mut arg_type);
+            let arg_name = &arg.pat;
+            let arg_name_str = quote! { #arg_name }.to_string();
+            Some(quote! {
+                event_abi.add_input::<#arg_type>(#arg_name_str, #indexed);
+                contract_abi.add_type_descriptions::<#arg_type>();
+            })
+        })
+        .collect();
+
+    quote! {
+        let mut event_abi = elrond_wasm::abi::EventAbi{
+            docs: &[ #(#event_docs),* ],
+            identifier: #event_name,
+            inputs: elrond_wasm::types::heap::Vec::new(),
+        };
+        #(#input_snippets)*
+    }
+}
+
+fn generate_event_snippets(contract: &ContractTrait) -> Vec<proc_macro2::TokenStream> {
+    contract
+        .methods
+        .iter()
+        .filter_map(|m| match &m.public_role {
+            PublicRole::Event(event_metadata) => {
```

Comment:
Replace with something like:
`if let MethodImpl::Generated(AutoImpl::Event{identifier}) = m.implementation { ... `

Event is not a PublicRole, since it is not publicly callable.
... maybe we should rename PublicRole to PubliclyCallable to avoid future confusions